src/Raven.Server/Documents/TimeSeries/TimeSeriesReader.cs

```diff
@@ -442,13 +450,14 @@ public class TimeSeriesMultiReader : ITimeSeriesReader, IEnumerator<TimeSeriesRe
         private readonly string _collection;
         private readonly TimeSpan? _offset;
         private readonly TimeValue _groupBy;
+        private readonly CancellationToken _token;
         private readonly DateTime _from, _to;
         private Stack<(DateTime Start, string Name)> _timeseriesStack;
 
         public bool IsRaw => _reader.IsRaw;
 
         public TimeSeriesMultiReader(DocumentsOperationContext context, string documentId,
-            string source, string collection, DateTime from, DateTime to, TimeSpan? offset, TimeValue groupBy)
+            string source, string collection, DateTime from, DateTime to, TimeSpan? offset, TimeValue groupBy, CancellationToken token = default)
```

Comment:
is there a place where are you not passing a token here? it is a potential place where you can introduce by accident a bug if this value has a default